### PR TITLE
update grafana deploy name

### PR DESCRIPTION
**Pull request by songleo**
due to we change grafana deployment name. this breaks our grafana-dev feature. so we need to update the script.
/hold

### CODE REVIEW

**Comment by songleo** [comment]
need to fix this error:

```
$ ./switch-to-grafana-admin.sh kube:admin
User <kube:admin> switched to be grafana admin
command terminated with exit code 1
Failed to disable getting start for the user <kube:admin>
```

**Comment by clyang82** [comment]
I have draft code diff for this fix, you can apply into your PR
```
diff --git a/tools/setup-grafana-dev.sh b/tools/setup-grafana-dev.sh
index 3c8339b..90006ec 100755
--- a/tools/setup-grafana-dev.sh
+++ b/tools/setup-grafana-dev.sh
@@ -7,6 +7,9 @@ if [[ "$(uname)" == "Darwin" ]]; then
     sed_command='sed -i '-e' -e'
 fi

+obs_namespace='open-cluster-management-observability'
+obs_name='observability'
+
 usage() {
   cat <<EOF
 Usage: $(basename "${BASH_SOURCE[0]}") [-h] [-d | -c]
@@ -23,27 +26,27 @@ EOF
 }

 deploy() {
-  kubectl get secret -n open-cluster-management-observability grafana-config -o 'go-template={{index .data "grafana.ini"}}' | base64 --decode > grafana-dev-config.ini
+  kubectl get secret -n $obs_namespace grafana-config -o 'go-template={{index .data "grafana.ini"}}' | base64 --decode > grafana-dev-config.ini
   if [ $? -ne 0 ]; then
       echo "Failed to get grafana config secret"
       exit 1
   fi
   $sed_command "s~%(domain)s/grafana/$~%(domain)s/grafana-dev/~g" grafana-dev-config.ini
-  kubectl create secret generic grafana-dev-config -n open-cluster-management-observability --from-file=grafana.ini=grafana-dev-config.ini
+  kubectl create secret generic grafana-dev-config -n $obs_namespace --from-file=grafana.ini=grafana-dev-config.ini

-  kubectl get deployment -n open-cluster-management-observability -l app=multicluster-observability-grafana -o yaml > grafana-dev-deploy.yaml
+  kubectl get deployment -n $obs_namespace -l app=multicluster-observability-grafana -o yaml > grafana-dev-deploy.yaml
   if [ $? -ne 0 ]; then
       echo "Failed to get grafana deployment"
       exit 1
   fi
-  $sed_command "s~name: grafana$~name: grafana-dev~g" grafana-dev-deploy.yaml
+  $sed_command "s~name: ${obs_name}-grafana$~name: grafana-dev~g" grafana-dev-deploy.yaml
   $sed_command "s~replicas:.*$~replicas: 1~g" grafana-dev-deploy.yaml
   $sed_command "s~grafana-config$~grafana-dev-config~g" grafana-dev-deploy.yaml
   $sed_command "s~app: multicluster-observability-grafana$~app: multicluster-observability-grafana-dev~g" grafana-dev-deploy.yaml
   $sed_command "s~grafana-config$~grafana-dev-config~g" grafana-dev-deploy.yaml

-  POD_NAME=$(kubectl get pods -n open-cluster-management-observability|grep grafana|awk '{split($0, a, " "); print a[1]}' |head -n 1)
-  GROUP_ID=$(kubectl exec -n open-cluster-management-observability $POD_NAME -c grafana -- bash -c "ls -l /var/lib/grafana/grafana.db"|awk '{split($0, a, " "); print a[4]}')
+  POD_NAME=$(kubectl get pods -n $obs_namespace|grep grafana|awk '{split($0, a, " "); print a[1]}' |head -n 1)
+  GROUP_ID=$(kubectl exec -n $obs_namespace $POD_NAME -c grafana -- bash -c "ls -l /var/lib/grafana/grafana.db"|awk '{split($0, a, " "); print a[4]}')
   if [[ ${GROUP_ID} == "grafana" ]]; then
     GROUP_ID=472
   fi
@@ -52,7 +55,7 @@ deploy() {
   tr $ '\n' < grafana-dev-deploy.yaml.bak > grafana-dev-deploy.yaml
   kubectl apply -f grafana-dev-deploy.yaml

-  kubectl get svc -n open-cluster-management-observability -l app=multicluster-observability-grafana -o yaml > grafana-dev-svc.yaml
+  kubectl get svc -n $obs_namespace -l app=multicluster-observability-grafana -o yaml > grafana-dev-svc.yaml
   if [ $? -ne 0 ]; then
       echo "Failed to get grafana service"
       exit 1
@@ -65,7 +68,7 @@ deploy() {
   $sed_command 's/\- [0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}//g' grafana-dev-svc.yaml
   kubectl apply -f grafana-dev-svc.yaml

-  kubectl get ingress -n open-cluster-management-observability grafana -o yaml > grafana-dev-ingress.yaml
+  kubectl get ingress -n $obs_namespace grafana -o yaml > grafana-dev-ingress.yaml
   if [ $? -ne 0 ]; then
       echo "Failed to get grafana ingress"
       exit 1
@@ -80,7 +83,7 @@ kind: PersistentVolumeClaim
 apiVersion: v1
 metadata:
   name: grafana-dev
-  namespace: open-cluster-management-observability
+  namespace: ${obs_namespace}
 spec:
   accessModes:
     - ReadWriteOnce
@@ -89,7 +92,7 @@ spec:
       storage: 1Gi
   storageClassName: gp2
 EOL
-  PVC_CLASS=$(kubectl get pvc -n open-cluster-management-observability alertmanager-db-alertmanager-0 -o yaml|grep "  storageClassName")
+  PVC_CLASS=$(kubectl get pvc -n $obs_namespace alertmanager-db-$obs_name-alertmanager-0 -o yaml|grep "  storageClassName")
   $sed_command "s~  storageClassName:.*$~${PVC_CLASS}~g" grafana-pvc.yaml
   kubectl apply -f grafana-pvc.yaml

@@ -98,11 +101,11 @@ EOL
 }

 clean() {
-  kubectl delete secret -n open-cluster-management-observability grafana-dev-config
-  kubectl delete deployment -n open-cluster-management-observability grafana-dev
-  kubectl delete svc -n open-cluster-management-observability grafana-dev
-  kubectl delete ingress -n open-cluster-management-observability grafana-dev
-  kubectl delete pvc -n open-cluster-management-observability grafana-dev
+  kubectl delete secret -n $obs_namespace grafana-dev-config
+  kubectl delete deployment -n $obs_namespace grafana-dev
+  kubectl delete svc -n $obs_namespace grafana-dev
+  kubectl delete ingress -n $obs_namespace grafana-dev
+  kubectl delete pvc -n $obs_namespace grafana-dev
 }

 msg() {
```

**Comment by clyang82** [comment]
BTW, please provide two parameters `--name` is for CR name and `--namespace` is for MCO namespace.

**Comment by clyang82** [comment]
Another consideration is can we write a test case to install the grafana-dev so that we can avoid such break.

File: tools/setup-grafana-dev.sh [hunk 1]
```diff
@@ -89,7 +101,7 @@ spec:
       storage: 1Gi
   storageClassName: gp2
 EOL
-  PVC_CLASS=$(kubectl get pvc -n open-cluster-management-observability alertmanager-db-alertmanager-0 -o yaml|grep "  storageClassName")
+  PVC_CLASS=$(kubectl get pvc -n "$obs_namespace" alertmanager-db-observability-alertmanager-0 -o yaml|grep "  storageClassName")
```
Comment (clyang82) [inline tools/setup-grafana-dev.sh:104]: s/observability/$obs_cr_name

**Pull request merged**